packages/idealo.design/scripts/motifuiImporter.js

```diff
@@ -29,62 +21,134 @@ async function readDirectory(directory) {
-
-        res.on('data', d => {
-            process.stdout.write(d)
+async function storeScreenshotFolderName(components) {
+    for(const component of components){
+        await fs.promises.readFile(component.pathToStoryFile, 'utf8').then((contentOfStoryFile) => {
+            if (contentOfStoryFile.includes('const stories = storiesOf(')) {
+                const startIndex = contentOfStoryFile.indexOf('storiesOf(')
+                const endIndex = contentOfStoryFile.indexOf(', module')
+                component.screenshotFolderName = contentOfStoryFile.substring(endIndex - 1, startIndex + 11)
+                delete component.pathToStoryFile
+            }
         })
-    })
-
-    req.on('error', error => {
-        console.error(error)
-    })
-
-    req.write(JSON.stringify(data))
-    req.end()
+    }
+    return components
 }
 
-async function handleImportProcess(source, destination){
-    await fse.remove(destination)
-        .then(()=> console.log('delete folder successfully'))
-        .catch(err => console.log(err))
-    await fse.copy(source, destination)
-        .then(()=>console.log('copy folder successfully'))
-        .catch(err => console.log(err))
-    readDirectory(destination)
-        .then((result)=>readPackageJsons(result, destination)).then((result)=>sendDataToHttpRequest(result))
+async function storePathToScreenshots(components){
+    for(const component of components){
+        await fs.promises.readdir(localScreenshots + '/' + component.screenshotFolderName).then((screenshots) => {
+            component.screenshots = screenshots
+        })
+    }
+    return components
 }
 
-if(dangerousUpdateModeArgument){
-    handleImportProcess(pathToMotifUiRepo, motifUiFolder).then(()=>console.log('process finished'))
-}else{
-    console.error('something went wrong')
-}
+async function createFormDataForComponent(components) {
+    for (const component of components) {
+
+        await fs.mkdirSync(localPathToMotifUIScreenshots+'/'+component.screenshotFolderName, (err) => {
+            if (err) {
+                throw err;
+            }
+        });
+        const componentFormData = new FormData();
 
-module.exports.modeArgument = dangerousUpdateModeArgument
+        componentFormData.append('name', component.name);
+        for (const keyword of component.keywords) {
+            componentFormData.append('keywords', keyword);
+        }
 
+        componentFormData.append('screenshotFolderName', component.screenshotFolderName);
+        componentFormData.append('readme', component.readme);
 
+        for (const screenshot of component.screenshots) {
+            const screenshotBuffer = await fs.readFileSync(localScreenshots + '/' + component.screenshotFolderName + '/' + screenshot);
+            const screenshotName= screenshot.replace(/ /g, "_");
+            componentFormData.append('screenshots', screenshotBuffer, screenshotName);
+        }
+        await sendDataToHttpRequest(componentFormData)
```

Comment:
Please be careful with mixing concerns. The function is called `create form data` but in the end it also sends it. Maybe just return the form data and then call `send` explicitly in the `handleImportProcess` function?

Reply:
This makes sense. We'll do that. Thank you!